internal/metrics/db.go

```diff
@@ -76,33 +75,85 @@ func FlushMetrics(ctx context.Context) error {
 	return nil
 }
 
-// / Spens up to 15 seconds sending all metrics collected so far to flyctl-metrics post endpoint
-func SendMetrics(ctx context.Context, json string) error {
-	authToken, err := GetMetricsToken(ctx)
+func SendMetrics(ctx context.Context, jsonData string) error {
+	cfg := config.FromContext(ctx)
+	metricsToken, err := GetMetricsToken(ctx)
 	if err != nil {
-		return err
+		fmt.Fprintf(os.Stderr, "Warning: Metrics token unavailable: %v\n", err)
+		return nil
 	}
 
-	cfg := config.FromContext(ctx)
-	request, err := http.NewRequest("POST", cfg.MetricsBaseURL+"/metrics_post", bytes.NewBuffer([]byte(json)))
+	baseURL := cfg.MetricsBaseURL
+	endpoint := baseURL + "/metrics_post"
+	userAgent := fmt.Sprintf("flyctl/%s", buildinfo.Info().Version)
+
+	errChan := make(chan error, 1)
+
+	go sendMetricsRequest(endpoint, metricsToken, userAgent, []byte(jsonData), errChan)
+
+	err = waitForCompletion(errChan)
```

Comment:
Starting a goroutine and then waiting on it seems like a code smell. This could just be a synchronous call to `sendMetricsRequest()` and you could pass a `ctx` in that has a timeout instead.